src/features/profile/skills/add-skills/use-form.tsx

```diff
@@ -0,0 +1,26 @@
+import { useAuthState } from "hooks/use-auth-state";
+import { addProfileSkill } from "../../services/profile.service";
+import { FormEvent } from "react";
+
+const useForm = ({ setIsOpen, selectedSkillId }: useFormProps) => {
+  const { token } = useAuthState();
+  const onSubmitForm = async (event: FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    try {
+      if (!token || !selectedSkillId) return;
+      const skillId = parseInt(selectedSkillId);
+      const data = await addProfileSkill(token, { skillId });
+      setIsOpen(false);
+      return data;
+    } catch (error) {
+      console.log(error);
+    }
+  };
```

Comment:
Esto definitivamente debe ser un useCallback, recuerda tambien, cuando pasa una funcion como parametro de un componente por props debe ser si o si un useCallback

```suggestion
  const onSubmitForm = useCallback(async (event: FormEvent<HTMLFormElement>) => {
    event.preventDefault();
    try {
      if (!token || !selectedSkillId) return;
      const skillId = parseInt(selectedSkillId);
      const data = await addProfileSkill(token, { skillId });
      setIsOpen(false);
      return data;
    } catch (error) {
      console.log(error);
    }
  }, [...]);
```